models/topic.go

```diff
@@ -47,10 +47,16 @@ type Topic struct {
 	Links       *TopicLinks `bson:"links,omitempty"          json:"links,omitempty"`
 	ReleaseDate *time.Time  `bson:"release_date,omitempty"   json:"release_date,omitempty"`
 	State       string      `bson:"state,omitempty"          json:"state,omitempty"`
-	SubtopicIds []string    `bson:"subtopics_ids,omitempty"  json:"-"`
+	SubtopicIds []string    `bson:"subtopics_ids,omitempty"  json:"subtopics_ids,omitempty"`
 	Title       string      `bson:"title,omitempty"          json:"title,omitempty"`
 }
 
+// TopicRelease represents the incoming request structure containing a topic update
+type TopicUpdate struct {
```

Comment:
The PUT endpoint should update the whole resource; so if a field isn't present it removes it from the document; hence this endpoint should handle nearly all fields associated with this resource; otherwise we would want to implement a PATCH endpoint that only updates part of the resource (this then requires us to implement IF-Match and ETag header - increasing the work).

Reply:
Updated so it updates all fields